core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -521,16 +509,17 @@ private[spark] class BlockManager(
               }
```

Comment:
Yeah, I think this is okay since we're not mutating anything which is being concurrent read in this scenario.